src/parser/mod.rs

```diff
@@ -28,6 +28,7 @@ pub struct SvgElement {
     pub attributes: HashMap<String, String>,
     pub replacement: HashMap<String, String>,
     pub nodes: Vec<SvgElement>,
+    pub type_override: Option<String>,
```

Comment:
I think we should use an enum for that kind of handling. Enum can implement `Display` trait, it should simplify handling of logic afterwards

Reply:
I'm still quite new to Rust, thanks a lot for your advice!